src/Microsoft.OData.Core/BindingPathHelper.cs

```diff
@@ -43,7 +44,7 @@ public static bool MatchBindingPath(IEdmPathExpression bindingPath, List<ODataPa
                 // Containment navigation property or complex property in binding path.
                 if (segment is PropertySegment || (segmentIsNavigationPropertySegment && segment.TargetEdmNavigationSource is IEdmContainedEntitySet))
                 {
-                    if (pathIndex < 0 || string.CompareOrdinal(paths[pathIndex], segment.Identifier) != 0)

```

Comment:
>CompareOrdinal [](start = 48, length = 14)

I don't know which one is better. But from [https://www.dotnetjalps.com/2007/11/what-is-diffrenece-between.html](https://www.dotnetjalps.com/2007/11/what-is-diffrenece-between.html), it seems CompareOrdinal is more faster. (I don't know). 

But, I prefer to "CompareOrdinal", it's more readable, maintainable.

Reply:
After experiments, CompareOrdinal was faster.